XRTK-Core/Packages/com.xrtk.core/Tests/Services/ITestExtensionService1.cs

```diff
@@ -1,9 +1,30 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System.Collections.Generic;
 using XRTK.Interfaces;
 
 namespace XRTK.Tests.Services
 {
-    internal interface ITestExtensionService1 : IMixedRealityExtensionService, ITestService { }
+    internal interface ITestExtensionService1 : IMixedRealityExtensionService
+    {
+        /// <summary>
+        /// The list of <see cref="IMixedRealityExtensionDataProvider"/>s registered and running with the system.
+        /// </summary>
+        IReadOnlyCollection<IMixedRealityExtensionDataProvider> DataProviders { get; }
+
+        /// <summary>
+        /// Registers the <see cref="ITestExtensionDataProvider1"/> with the <see cref="ITestService"/>.
+        /// </summary>
+        /// <param name="dataProvider"></param>
+        /// <returns></returns>
+        bool RegisterDataProvider(ITestExtensionDataProvider1 dataProvider);
```

Comment:
Seems kinda off that this is here and not being properly registered via the xrtk service registry

Reply:
This is the implementation used by ALL of the current services in the XRTK, so the test was updated to reflect how it is currently implemented.

Please feel free to submit a PR and change the core systems and then the tests if you don't like it.

Reply:
Yeah but I don't think it's actually part of any interface, just a pattern we tend to use, which is why I asked. I think this might benefit from either adding to the service interface and/or base service class.

Reply:
Agreed there, I hate boilerplate if it can be genericized in the base somewhere for dataproviders to call.

Reply:
Needs to be addressed in a separate PR